typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,129 @@
+import * as bindingsJsGrpc from '../../kurtosis_core_rpc_api_bindings/api_container_service_grpc_pb'; //TODO - should I be importing all modules or just specific ones for best practices (need to check)
+import * as bindingsJs from '../../kurtosis_core_rpc_api_bindings/api_container_service_pb'; //TODO (comment) - I cannont merge with above line unfortunately, can only import one module at a time MAXIMUM
+import { ServiceID } from './service';
+import { StaticFileID } from './container_creation_config'; 
+import * as grpc from "grpc"; //TODO - this is needed for grpc callback, but this import is tentative 
+import * as constructors from "../constructor_calls";
+var path = require("path"); //TODO - I don't think this works
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+class GeneratedFileFilepaths { //TODO (comment) - try interface if class doesn't work
+	absoluteFilepathOnTestsuiteContainer: string;
```

Comment:
1) Typescript (like Java, and unlike Go) has a modifier for fields that declares that they can only be assigned once, in the constructor, and then can't be modified again. This is good - it makes our program more immutable (and I'm not happy Go doesn't have it).

That keyword is `readonly` in Typescript, and we should use it here.

Reply:
SECOND THING:
In the same vein, it's actually bad that I made these public fields in Go at all. Things should be private by default, with getters for accessing them if needed.

So this class (which is just a read-only container class) should actually have:
1) `private readonly` keywords on the fields
2) a constructor for initializing those fields exactly once
3) getters for getting the values

I should have actually done this for the Go version as well; I'll do so now